src/message/mailbox/parsers/rfc2822.rs

```diff
@@ -0,0 +1,146 @@
+//! Partial parsers implementation of [RFC2822]: Internet Message
+//! Format.
+//!
+//! [RFC2822]: https://datatracker.ietf.org/doc/html/rfc2822
+
+use chumsky::prelude::*;
+use once_cell::sync::Lazy;
+
+use super::{rfc2234, rfc5336};
+
+// 3.2.1. Primitive Tokens
+// https://datatracker.ietf.org/doc/html/rfc2822#section-3.2.1
+
+static TEXT: Lazy<Vec<char>> = Lazy::new(|| {
+    let mut range: Vec<char> = vec![];
+    range.extend((1 as char)..=(9 as char));
+    range.push(11 as char);
+    range.push(12 as char);
+    range.extend((14 as char)..=(127 as char));
+    range
+});
+
+// text            =       %d1-9 /         ; Characters excluding CR and LF
+//                         %d11 /
+//                         %d12 /
+//                         %d14-127 /
+//                         obs-text
+fn text() -> impl Parser<char, char, Error = Simple<char>> {
+    one_of(TEXT.as_slice())
+}
```

Comment:
```suggestion
// text            =       %d1-9 /         ; Characters excluding CR and LF
//                         %d11 /
//                         %d12 /
//                         %d14-127 /
//                         obs-text
fn text() -> impl Parser<char, char, Error = Simple<char>> {
    filter(|c| matches!(u32::from(*c), 1..=9 | 11 | 12 | 14..))
}
```

I think this is cleaner. Maybe we can do a quick benchmark and see if it is a lot slower. It is probably also easier for the optimizer to understand.

Note: I think we need `14..` not `14..=127` to deal with already-decoded Unicode. Do we have a test for that, or how was it working before?

Reply:
> I think this is cleaner. Maybe we can do a quick benchmark and see if it is a lot slower. It is probably also easier for the optimizer to understand.

I totally agree, I will change that.

> Note: I think we need `14..` not `14..=127` to deal with already-decoded Unicode. Do we have a test for that, or how was it working before?

No, because from the rfc5336:

```rust
// uqcontent = qcontent / UTF8-non-ascii
pub(super) fn uqcontent() -> impl Parser<char, char, Error = Simple<char>> {
    choice((rfc2822::qcontent(), utf8_non_ascii()))
}
```

ASCII part is managed by classical rfc2822, but non ASCII part (UTF8) are managed by rfc5336.

Reply:
:heavy_check_mark:

Reply:
Ah, I see. I think I was confused by this not being used by display-name but that is because we are parsing it in a funny conservative way. We'll discuss that in the other thread https://github.com/lettre/lettre/pull/839#discussion_r1064067658